src/geometry/types.rs

```diff
@@ -185,9 +186,23 @@ pub struct Size {
 }
 
 impl Size {
-    pub fn new(width: u32, height: u32) -> Size {
-        if width == 0 || height == 0 { panic!("A size's width and/or height must be positive."); }
-        Size { width, height }
+
+    pub fn new(exp: u32) -> Size {
+
+        //if exp < 0 { panic!("An Exponent must be positive! "); }
+        let width = u32::pow(2, exp);
+        let height = u32::pow(2, exp);
+
+        Size { width, height}
+    }
+
+    pub fn half_size(&self) -> Size {
+        let wd = self.width/2;
+        Size {width: wd, height: wd}
+    }
+
+    pub fn new_wh(width: u32, height: u32) -> Size {
+        Size{width, height}
     }
 }
```

Comment:
Is this method necessary? It makes it possible to create sizes with unequal width and height, as well as width and heights that aren't a power of 2, which is something we want to restrict.